lib/OpenQA/WebSockets/Controller/Worker.pm

```diff
@@ -172,62 +168,28 @@ sub _message {
                 # Tell the worker that we saw it (used for tests and debugging)
                 $tx->send({json => {type => 'info', seen => 1}});
             }
-            catch {
-                log_error("Failed updating seen and error status of worker $worker_id: $_");    # uncoverable statement
-            };
+            catch ($e) {
+                log_error("Failed updating seen and error status of worker $worker_id: $e");    # uncoverable statement
+            }
         }
 
         # find the job currently associated with that worker and check whether the worker still
         # executes the job it is supposed to
         my $current_job_id;
         try {
-            return undef unless $worker;
-
-            my $registered_job_token;
-            my $current_job_state;
-            my @unfinished_jobs = $worker->unfinished_jobs;
-            my $current_job = $worker->job // $unfinished_jobs[0];
-            if ($current_job) {
-                $current_job_id = $current_job->id;
-                $current_job_state = $current_job->state;
+            my ($current_job, $unfinished_jobs)
+              = $self->_should_assign_jobs($worker, $current_job_id, $worker_id, $job_id, $job_token,
+					   $worker_previously_idle);
+	    log_debug("Rescheduling jobs assigned to worker $worker_id");
+                
+            if (defined $unfinished_jobs) {
```

Comment:
```suggestion
            if (defined $unfinished_jobs) {
                log_debug("Rescheduling jobs assigned to worker $worker_id");
```

Reply:
I tried this because of the CI. which I couldnt reproduce local. I will revert